src/index.ts

```diff
@@ -50,11 +50,39 @@ export const temporal = (<TState>(
         temporalStateCreator(set, get, options),
     );
 
+    const handleSet = (
+      pastState: TState,
+      // `replace` will likely be deprecated and removed in the future
+      replace: boolean | undefined,
+      currentState: TState,
+      deltaState?: Partial<TState>,
+    ) => {
+      if (store.temporal.getState().isTracking) {
+        // This naively assumes that only one new state can be added at a time
+        if (
+          options?.limit &&
+          store.temporal.getState().pastStates.length >= options?.limit
+        ) {
+          store.temporal.getState().pastStates.shift();
+        }
+
+        (store.temporal.getState() as _TemporalState<TState>)._onSave?.(
+          pastState,
+          currentState,
+        );
+        store.temporal.setState({
+          ...store.temporal.getState(),
```

Comment:
Why is this line needed here? Shouldn't we only need to be passing `pastStates` and `futureStates` here?

Reply:
I (mistakenly, I think) was under the impression that `setState` did not do the kind of auto-merging of state that `set` does, which is why I spread the other values here. I will remove this line.

Reply:
Gotcha, thanks. I _think_ they both behave the same.